contracts/USDCPool.sol

```diff
@@ -12,163 +12,45 @@ contract USDCPool is
     ERC20("Cruize USDC LP Token", "writeUSDC")
 {
     using SafeMath for uint256;
-    uint256 public lockupPeriod = 2 weeks;
-    uint256 public lockedAmount;
-    uint256 public lockedPremium;
-    mapping(address => uint256) private lastProvideTimestamp;
-
+    address constant USDC_Token_Address = 0xb7a4F3E9097C08dA09517b5aB877F7a917224ede;
     /*
      * @nonce Sends premiums to the liquidity pool
      **/
     receive() external payable {}
-
-    /**
-     * @notice Used for changing the lockup period
-     * @param value New period value
-     */
-    function setLockupPeriod(uint256 value) external override onlyOwner {
-        require(value <= 60 days, "Lockup period is too large");
-        lockupPeriod = value;
-    }
-
-    /*
+      /*
      * @nonce A provider supplies USDC to the pool and receives writeUSDC tokens
      * @param minMint Minimum amount of tokens that should be received by a provider.
-                      Calling the provide function will require the minimum amount of tokens to be minted.
-                      The actual amount that will be minted could vary but can only be higher (not lower) than the minimum value.
+      Calling the provide function will require the minimum amount of tokens to be minted.
+      The actual amount that will be minted could vary but can only be higher (not lower) than the minimum value.
      * @return mint Amount of tokens to be received
      */
-    function provide(uint256 minMint) external payable returns (uint256 mint) {
-        // we will send form the front
-        require(msg.value > 0,"Pool: Amount is too small");
-        lastProvideTimestamp[msg.sender] = block.timestamp;
-        uint supply = totalSupply();
-        uint balance = totalBalance();
-        if (supply > 0 && balance > 0)
-            mint = msg.value.mul(supply).div(balance.sub(msg.value));
-        else
-            mint = msg.value.mul(1000);
 
-        require(mint >= minMint, "Pool: Mint limit is too large");
-     
+    function provide(uint256 USDC_Count) external payable returns (uint256) {
```

Comment:
`why we made this function payable, if we are not receiving ethers in our function then there is no need to make it payable, because we are not handling ethers in this function`